axiom/nr_log_event.h

```diff
@@ -51,7 +67,23 @@ extern char* nr_log_event_to_json(const nr_log_event_t* event);
  * Returns : True on success; false on error.
  */
 extern bool nr_log_event_to_json_buffer(const nr_log_event_t* event,
-                                         nrbuf_t* buf);
+                                        nrbuf_t* buf);
+
+/*
+ * Purpose : Append New Relic format JSON for a log event to a buffer.
+ *
+ * Params  : 1. The log event.
+ *           2. The buffer to append to.
+ *           3. Whether or not to wrap JSON with starting and
+ *              ending "[" "]".  If enabled the resulting
+ *              JSON is not valid unless enclosed in another
+ *              JSON string.
+ *
+ * Returns : True on success; false on error.
+ */
+extern bool nr_log_event_to_json_buffer_ex(const nr_log_event_t* event,
+                                           nrbuf_t* buf,
+                                           bool partial);
```

Comment:
The naming follows the naming scheme used for span events - I would prefer to retain the name as it is specific to the type of serialization occurring.   If we supported multiple serialization methods I could see using a more general name.

The API also follows that used for span events (and other events) in that all other functions which generate JSON for events makes valid JSON strings including the wrapping '[' ']'.   To be consistent we provide similarly named functions for log events which provide valid JSON.  However when constructing a payload for the log event endpoint on the collector we cannot string together these valid JSON strings (like apparently works for other event types).  So for efficiency I created a special version of the function which could generate an invalid JSON string (but a valid JSON substring) if requested.

It might be the non-"ex" versions of these functions are not needed at all and perhaps I could leave just the "ex" version.  I'd like to designate it differently as there is precedence with the span event versions of these functions that a valid JSON string is returned.

Reply:
Ok. I am convinced about the reason for `nr_log_event_to_json_buffer`. But I'd still suggest reducing unused code by dropping the `_ex` suffix and `bool partial` arguments and remove the original `nr_log_event_to_json_buffer` that creates an array of one element.

Reply:
Fixed in commit [2622c75](https://github.com/newrelic/newrelic-php-agent/pull/515/commits/2622c757d16d109329f6bcc4b7b571cadbfcd4cf).

Reply:
Looks good. Thank you for bearing with me.